src/main/java/com/hubspot/jinjava/util/ScopeMap.java

```diff
@@ -19,6 +25,23 @@ public ScopeMap() {
   public ScopeMap(ScopeMap<K, V> parent) {
     this.scope = new HashMap<K, V>();
     this.parent = parent;
+
+    Set<ScopeMap<K, V>> parents = new HashSet<>();
```

Comment:
I think you only need to check new parent against the existing parents, right? If we're running this every time a new ScopeMap is created, it will get them all.

Reply:
So are you suggesting storing all the parents in a Set on the ScopeMap? Either way we have to run up the parent tree in the constructor.

Reply:
yeah, I was thinking just run up the whole parent tree in the constructor. I think that will take a trivial amount of time.

Reply:
Should I add a config to enable/disable this check?

Reply:
I can't think of a reason why you'd want to turn it off.